static/app/views/alerts/rules/metric/ruleForm.tsx

```diff
@@ -918,26 +919,39 @@ class RuleFormContainer extends DeprecatedAsyncComponent<Props, State> {
   };
 
   handleComparisonTypeChange = (value: AlertRuleComparisonType) => {
-    const comparisonDelta =
-      value === AlertRuleComparisonType.CHANGE
-        ? this.state.comparisonDelta ?? DEFAULT_CHANGE_COMP_DELTA
-        : undefined;
-    const timeWindow = this.state.comparisonDelta
-      ? DEFAULT_COUNT_TIME_WINDOW
-      : DEFAULT_CHANGE_TIME_WINDOW;
-    const sensitivity =
-      value === AlertRuleComparisonType.DYNAMIC
-        ? this.state.sensitivity || AlertRuleSensitivity.MEDIUM
-        : undefined;
-    const seasonality =
-      value === AlertRuleComparisonType.DYNAMIC ? AlertRuleSeasonality.AUTO : undefined;
-    this.setState({
-      comparisonType: value,
-      comparisonDelta,
-      timeWindow,
-      sensitivity,
-      seasonality,
-    });
+    this.setState({comparisonType: value});
+
+    switch (value) {
+      case AlertRuleComparisonType.DYNAMIC:
+        this.setState({
```

Comment:
these two set-states may collide since it's async, instead i'd recommend creating an `updateState` variable where everything is set to undefined / `comparisonType: value` then update each switch as needed. at the end invoke `setState(updateState)` once.